openlibrary/plugins/openlibrary/api.py

```diff
@@ -139,10 +139,14 @@ def GET(self, work_id):
         return simplejson.dumps(result)
 
     def POST(self, work_id):
+        """
+        We can't use the action because we do want to add the book if it doesn't exist yet. 
+        If the book already exists, we don't want to remove it.
+        """
```

Comment:
```suggestion
        """
        Add a work (or a work and an edition) to a bookshelf.
        
        GET params:
        - edition_id str (optional)
        - action str: e.g. "add", "remove"
        - redir bool: if patron not logged in, redirect back to page after login
        - bookshelf_id int: which bookshelf? e.g. the ID for "want to read"?
        - dont_remove bool: if action is "add" and book exists, use this to avoid removing it  

        :param str work_id: e.g. OL123W
        :rtype: json
        :return: a list of bookshelves_affected
        """
```